src/server/handlers/launchpad.js

```diff
@@ -437,6 +439,21 @@ export const findSnap = (req, res) => {
     .catch((error) => sendError(res, error));
 };
 
+const clearSnapCache = (repositoryUrl) => {
+  const repository = parseGitHubUrl(repositoryUrl);
+  const enabledReposCacheId = getUrlPrefixCacheId(getRepoUrlPrefix(repository.owner));
+  const snapCacheId = getRepositoryUrlCacheId(repositoryUrl);
+  const snapNameCacheId = getSnapNameCacheId(repositoryUrl);
+
+  logger.info(`Clearing caches for ${repositoryUrl}: ${enabledReposCacheId}, ${snapCacheId}, ${snapNameCacheId}`);
+
+  return Promise.all([
+    getMemcached().del(enabledReposCacheId),
+    getMemcached().del(snapCacheId),
+    getMemcached().del(snapNameCacheId)
+  ]);
+};
```

Comment:
I think snap names are cached twice, once on their own and again with the rest of the snap data. Is it worth consolidating these cache items?